lib/rex/proto/rfb/cipher.rb

```diff
@@ -83,14 +83,30 @@ def self.encrypt_ard(username, password, generator, key_length, prime_modulus, p
 
     user_struct = username + ("\0" * (64 - username.length)) + password + ("\0" * (64 - password.length))
 
-    dh_peer = OpenSSL::PKey::DH.new(key_length * 8, generator)
-    dh_peer.set_key(peer_public_key, nil)
-
-    dh = OpenSSL::PKey::DH.new(dh_peer)
-    dh.set_pqg(prime_modulus, nil, generator)
-    dh.generate_key!
-
-    shared_key = dh.compute_key(dh_peer.pub_key)
+    # OpenSSL 3.0+
```

Comment:
Code path triggered via vnc_login and connecting to an apple VNC server after manually adding a user that has permissions to log in, as well as going into Mac's sharing menu, enabling `Screen Sharing` and granting access to the new `test_user` account

```
msf6 auxiliary(scanner/vnc/vnc_login) > run 192.168.123.1 username=test_user password=p4$$w0rd1

[*] 192.168.123.1:5900    - 192.168.123.1:5900 - Starting VNC login sweep
[!] 192.168.123.1:5900    - No active DB -- Credential data will not be saved!
[+] 192.168.123.1:5900    - 192.168.123.1:5900 - Login Successful: test_user:p4$$w0rd1
[*] 192.168.123.1:5900    - Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```